src/geometryRTheta/time_solver/bsl_predcorr.hpp

```diff
@@ -123,74 +123,87 @@ class BslPredCorrRTheta : public ITimeSolverRTheta
+                charge_density(get_const_field(density_coef_host), m_spline_evaluator);
+
+        // Setup
+        m_builder(get_field(density_coef_host), get_const_field(density_host));
+        m_poisson_solver(charge_density, electrical_potential);
+        ddc::parallel_deepcopy(electrical_potential_host, get_const_field(electrical_potential));
         ddc::PdiEvent("iteration")
                 .with("iter", 0)
                 .with("time", 0)
                 .with("density", density_host)
-                .with("electrical_potential", electrical_potential0_host);
-
-
-        std::function<void(host_t<DVectorFieldRTheta<X, Y>>, host_t<DConstFieldRTheta>)>
-                define_advection_field = [&](host_t<DVectorFieldRTheta<X, Y>> advection_field_host,
-                                             host_t<DConstFieldRTheta> density_host) {
-                    // --- compute electrostatic potential:
-                    host_t<Spline2DMem> density_coef(get_spline_idx_range(m_builder));
-                    m_builder(get_field(density_coef), get_const_field(density_host));
-                    m_poisson_solver(charge_density_coord, get_field(electrostatic_potential_coef));
-
-                    // --- compute advection field:
-                    advection_field_computer(
-                            get_field(electrostatic_potential_coef),
-                            advection_field_host);
-                };
-
-        std::function<void(host_t<DFieldRTheta>, host_t<DConstVectorFieldRTheta<X, Y>>, double)>
-                advect_density = [&](host_t<DFieldRTheta> density_host,
-                                     host_t<DConstVectorFieldRTheta<X, Y>> advection_field_host,
-                                     double dt) {
-                    auto density = ddc::create_mirror_view_and_copy(
-                            Kokkos::DefaultExecutionSpace(),
-                            density_host);
-                    auto advection_field = ddcHelper::create_mirror_view_and_copy(
-                            Kokkos::DefaultExecutionSpace(),
-                            advection_field_host);
-                    m_advection_solver(get_field(density), get_const_field(advection_field), dt);
-                    ddc::parallel_deepcopy(density_host, density);
-                };
-
-        RK2<host_t<DFieldMemRTheta>,
-            host_t<DVectorFieldMemRTheta<X, Y>>,
-            Kokkos::DefaultHostExecutionSpace>
+                .with("electrical_potential", electrical_potential_host);
+
+        // RK2 methods
+        std::function<void(DVectorFieldRTheta<X, Y>, DConstFieldRTheta)> define_advection_field
+                = [&](DVectorFieldRTheta<X, Y> advection_field, DConstFieldRTheta density) {
+                      ddc::parallel_deepcopy(density_host, density);
+                      // --- compute electrostatic potential:
+                      m_builder(get_field(density_coef_host), get_const_field(density_host));
```

Comment:
```suggestion
                      m_builder(density_coef_host, get_const_field(density_host));
```
? 
or change for `density_coef_alloc_host`.

Reply:
[c246b3e](https://github.com/gyselax/gyselalibxx/pull/395/commits/c246b3e9a7accf6fda3a0a7e714d52383fdceaab)